src/kOS/Suffixed/OrbitInfo.cs

```diff
@@ -109,33 +111,33 @@ public OrbitableVelocity GetVelocityAtUT( TimeSpan timeStamp )
         }
 
         /// <summary>
-        /// Return the next OrbitInfo after this one (i.e. transitional encounter)
+        /// Returns the OrbitEta structure associated with this orbit, creating it if needed.
         /// </summary>
-        /// <returns>an OrbitInfo, or a null if there isn't any.</returns>
-        private OrbitInfo GetNextPatch()
+        /// <returns>an OrbitEta structure.</returns>
+        private OrbitEta GetETA ()
         {
-            if (GetHasNextPatch())
-            {
-                return new OrbitInfo(orbit.nextPatch, Shared);
+            if (eta == null) {
+                eta = new OrbitEta(orbit, Shared);
             }
-            throw new KOSSituationallyInvalidException("Cannot get next patch when no additional patches exist.  Try checking the HASNEXTPATCH suffix.");
```

Comment:
Man, git's diff output can really make things confusing at times.  It looked to me like you had replaced GetNextPatch with GetETA, which made no sense at all.  It took a while to realize that GetNextPatch was just pushed down a few lines in the file by GetETA, rather than replaced by it as the diff makes it look.

Reply:
Yeah, that's really not the greatest diff I've ever seen.

BTW, thanks to your comment, I noticed that I'd left in some unnecessary commit changes from an earlier iteration. Let me undo those and see if the diff gets more readable too.

Reply:
Turns out moving the new `GetETA` method earlier in the file so that there's enough other code in between makes diff actually realize that it's a completely new method. :)